src/common/zlib.cpp

```diff
@@ -1,118 +1,215 @@
+#   elif defined(_MSC_VER)
+#       define bswap16 _byteswap_ushort
+#       define bswap32 _byteswap_ulong
+#       define bswap64 _byteswap_uint64
+#   else
+#       error "No compiler builtins for byteswap available"
+#   endif
+#endif
+
+// Resolve the next address in jump table (0 == no jump, 1 == next address)
+#define JMPBIT(table, i) ((table[i / 8] >> (i & 7)) & 1)
+
+struct zlib_jump {
+    const void *ptr;
+};
+
+struct zlib {
+    std::vector<uint32> enc;
+    std::vector<struct zlib_jump> jump;
+};
+
+static struct zlib zlib;
+
+static void swap32_if_be(uint32 *v, const size_t memb)
 {
-    memset(zlib_compress_table, 0, sizeof(zlib_compress_table));
-    memset(zlib_decompress_table, 0, sizeof(zlib_decompress_table));
-
-    auto fp = fopen("compress.dat", "rb");
-    if (fp == NULL)
-        ShowFatalError("zlib_init: can't open file <compress.dat> \n");
-    fread(zlib_compress_table, sizeof(uint32), 512, fp);
-    fclose(fp);
-
-    uint32 temp_decompress_table[2556];
-    fp = fopen("decompress.dat", "rb");
-    if (fp == NULL)
-        ShowFatalError("zlib_init: can't open file <decompress.dat> \n");
-    fseek(fp, 0, SEEK_END);
-    auto size = ftell(fp);
-    fseek(fp, 0, SEEK_SET);
-    fread(temp_decompress_table, sizeof(char), size, fp);
-    fclose(fp);
-
-    // Align the jump table with our internal table..
-    for (auto x = 0; x < size / 4; x++)
-    {
-        if (temp_decompress_table[x] > 0xff)
-            zlib_decompress_table[x] = (uintptr)((uintptr*)zlib_decompress_table + ((temp_decompress_table[x] - 0x15b3aaa0) / 4));
-        else
-            zlib_decompress_table[x] = temp_decompress_table[x];
+#if DSP_BIG_ENDIAN
+    for (size_t i = 0; i < memb; ++i)
+        v[i] = bswap32(v[i]);
+#else
+    (void)v, (void)memb;
+#endif
+}
+
+static bool read_to_vector(const std::string &file, std::vector<uint32> &vec)
+{
+    FILE *f;
+    if (!(f = fopen(file.c_str(), "rb"))) {
+        ShowFatalError("zlib: can't open file <%s>\n", file.c_str());
+        return false;
     }
 
+    fseek(f, 0, SEEK_END);
+    const size_t size = ftell(f);
+    fseek(f, 0, SEEK_SET);
+
+    vec.resize(size / sizeof(uint32));
+    fread(vec.data(), sizeof(uint32), vec.size(), f);
+    fclose(f);
+
+    swap32_if_be(vec.data(), vec.size());
+    return true;
+}
+
+static void populate_jump_table(std::vector<struct zlib_jump> &jump, const std::vector<uint32> &dec)
+{
+    jump.resize(dec.size());
+
+    // Base address of dec table, if we substract pointer in dec table, we should should be
+    // able to normalize them to offsets starting from 0.
+    const uint32 base = dec[0] - sizeof(uint32);
+
+    for (size_t i = 0; i < dec.size(); ++i) {
+        if (dec[i] > 0xff) {
+            // Everything over 0xff are pointers.
+            // These pointers will be traversed until we hit data.
+            jump[i].ptr = jump.data() + (dec[i] - base) / sizeof(base);
+        } else {
```

Comment:
Thanks, will fix tomorrow. I recall there being style guide now too, I'll go read that _again_ :)